packages/rsocket-core/src/__tests__/RSocketClient-test.js

```diff
@@ -1225,7 +1225,38 @@ describe('RSocketClient', () => {
       });
     });
 
-    describe('metadataPush()', () => {});
+    describe('metadataPush()', () => {
+      // -> open
+      it('sends the payload', () => {
+        socket.metadataPush(payload).subscribe(subscriber);
+        expect(transport.sendOne.mock.calls.length).toBe(1);
+        expect(transport.sendOne.mock.frame).toEqual({
+          type: FRAME_TYPES.METADATA_PUSH,
+          flags: 0,
+          metadata: payload.metadata,
+          streamId: 0,
+        });
+        expect(subscriber.onComplete.mock.calls.length).toBe(1);
+        expect(subscriber.onError.mock.calls.length).toBe(0);
+      });
+
+      it('sends the payload with serialized data', () => {
+        ({socket, transport} = createSocket(JsonSerializers));
+        payload = {
+          metadata: {metadata: true},
+        };
+        socket.metadataPush(payload).subscribe(subscriber);
+        expect(transport.sendOne.mock.calls.length).toBe(1);
+        expect(transport.sendOne.mock.frame).toEqual({
+          type: FRAME_TYPES.METADATA_PUSH,
+          flags: 0,
+          metadata: '{"metadata":true}',
+          streamId: 0,
+        });
+        expect(subscriber.onComplete.mock.calls.length).toBe(1);
+        expect(subscriber.onError.mock.calls.length).toBe(0);
+      });
+    });
```

Comment:
It might also be future-proof to add a test for receiving metadata push frames as well, for instance 
```patch
Index: packages/rsocket-core/src/__tests__/RSocketClient-test.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/rsocket-core/src/__tests__/RSocketClient-test.js b/packages/rsocket-core/src/__tests__/RSocketClient-test.js
--- a/packages/rsocket-core/src/__tests__/RSocketClient-test.js	(revision d5e392c5745a7d4c5e2fb3949a288a3bb0da597d)
+++ b/packages/rsocket-core/src/__tests__/RSocketClient-test.js	(date 1615031729418)
@@ -324,7 +324,7 @@
     let socket;
     let subscriber;
 
-    function createSocket(serializers) {
+    function createSocket(serializers, responder) {
       const transport = genMockConnection();
       const client = new RSocketClient({
         serializers,
@@ -335,6 +335,7 @@
           metadataMimeType: '<metadataMimeType>',
         },
         transport,
+        responder,
       });
       let socket;
       client.connect().subscribe({
@@ -1256,6 +1257,23 @@
         expect(subscriber.onComplete.mock.calls.length).toBe(1);
         expect(subscriber.onError.mock.calls.length).toBe(0);
       });
+
+      it('receives the payload', (done) => {
+        const metadata = {'metadata': true};
+        ({socket, transport} = createSocket(JsonSerializers, {
+          metadataPush(payload) {
+            expect(payload.metadata).toEqual(metadata)
+            done();
+          }
+        }));
+        const metadataFrame = {
+          type: FRAME_TYPES.METADATA_PUSH,
+          flags: 0,
+          metadata: JsonSerializers.metadata.serialize(metadata),
+          streamId: 0,
+        };
+        transport.receive.mock.publisher.onNext(metadataFrame);
+      });
     });
 
     describe('client.close()', () => {
```

Reply:
done